hedera-node/hedera-app/src/test/java/com/hedera/node/app/blocks/impl/streaming/BlockNodeConnectionTest.java

```diff
@@ -0,0 +1,227 @@
+        blockNodeConnection.establishStream();
+
+        final var captor = ArgumentCaptor.forClass(StreamObserver.class);
+        verify(grpcServiceClient).bidi(any(), captor.capture());
+        final var capturedObserver = captor.getValue();
+        assertNotNull(capturedObserver);
+
+        final var response = PublishStreamResponse.newBuilder()
+                .setStatus(PublishStreamResponse.EndOfStream.newBuilder()
+                        .setStatus(PublishStreamResponseCode.STREAM_ITEMS_TIMEOUT)
+                        .setBlockNumber(1234)
+                        .build())
+                .build();
+        capturedObserver.onNext(response);
+
+        assertThat(logCaptor.infoLogs())
+                .contains(
+                        "Error returned from block node at block number 1234: status: STREAM_ITEMS_TIMEOUT\nblock_number: 1234");
+    }
+
+    @Test
+    void testGrpcClientStreamObserver_OnCompleted() {
+        when(nodeConfig.address()).thenReturn("localhost");
+        when(nodeConfig.port()).thenReturn(12345);
+        when(grpcServiceClient.bidi(any(), any(StreamObserver.class))).thenReturn(requestObserver);
+        blockNodeConnection.establishStream();
+
+        final var captor = ArgumentCaptor.forClass(StreamObserver.class);
+        verify(grpcServiceClient).bidi(any(), captor.capture());
+        final var capturedObserver = captor.getValue();
+        assertNotNull(capturedObserver);
+
+        capturedObserver.onCompleted();
+
+        assertThat(logCaptor.infoLogs()).contains("Stream completed for block node localhost:12345");
+        assertFalse(blockNodeConnection.isActive());
+        verify(blockNodeConnectionManager, times(1)).handleConnectionError(nodeConfig);
+    }
+
+    @Test
+    void testGrpcClientStreamObserver_OnError_StatusAborted() {
+        when(nodeConfig.address()).thenReturn("localhost");
+        when(nodeConfig.port()).thenReturn(12345);
+        when(grpcServiceClient.bidi(any(), any(StreamObserver.class))).thenReturn(requestObserver);
+
+        blockNodeConnection.establishStream();
+
+        final var captor = ArgumentCaptor.forClass(StreamObserver.class);
+        verify(grpcServiceClient).bidi(any(), captor.capture());
+
+        final var capturedObserver = captor.getValue();
+        assertNotNull(capturedObserver);
+
+        capturedObserver.onError(new StatusRuntimeException(Status.ABORTED));
+
+        assertThat(logCaptor.errorLogs())
+                .matches(
+                        logs -> logs.getFirst()
+                                .startsWith(
+                                        "Error in block node stream localhost:12345: Status{code=ABORTED, description=null, cause=null} io.grpc.StatusRuntimeException: ABORTED"));
+        assertFalse(blockNodeConnection.isActive());
+        verify(blockNodeConnectionManager, times(1)).handleConnectionError(nodeConfig);
+        verify(blockNodeConnectionManager, times(1)).scheduleReconnect(blockNodeConnection);
```

Comment:
```suggestion
        verify(blockNodeConnectionManager).handleConnectionError(nodeConfig);
        verify(blockNodeConnectionManager).scheduleReconnect(blockNodeConnection);
```
Mockito already complains if the method is called more than once